python/tests/catalog/test_jdbc.py

```diff
@@ -0,0 +1,36 @@
+import pytest
+
+from pyiceberg.catalog.jdbc import JDBCCatalog
+from pyiceberg.exceptions import NoSuchPropertyException
```

Comment:
There are a lot of good tests in Java's `CatalogTests`. It probably makes sense to copy those over here.

Reply:
Yes, it was on my todo list.  I also argued with Fokko that we should have a suite of integration tests that applies to all pyiceberg catalogs.  I ended up copying most of the tests from DynamoDB and Glue with not many changes, so I think it is possible to achieve.  This would ensure that all catalogs behave the same (which wasn't the case for Glue, see [issue](https://github.com/apache/iceberg/issues/7935)) and it would make contributing a new catalog that much easier to test.